src/python/pants/backend/scala/compile/scalac.py

```diff
@@ -48,6 +49,20 @@ class CompileScalaSourceRequest(ClasspathEntryRequest):
     field_sets_consume_only = (JavaFieldSet, JavaGeneratorFieldSet)
 
 
+@dataclass(frozen=True)
+class ScalacRequest:
```

Comment:
Should this include command-line flags as well? Whatever produces would fill that in from the applicable subsystem.

Reply:
Also maybe implement `EngineAwareParameter`?